core/src/com/unciv/logic/map/MapGenerator.kt

```diff
@@ -372,16 +372,23 @@ class MapGenerator(val ruleset: Ruleset) {
     // Here, we need each specific resource to be spread over the map - it matters less if specific resources are near each other
     private fun spreadStrategicResources(tileMap: TileMap, distance: Int) {
         val strategicResources = ruleset.tileResources.values.filter { it.resourceType == ResourceType.Strategic }
-        val totalNumberOfResources = tileMap.values.count { it.isLand && !it.getBaseTerrain().impassable } *
-                tileMap.mapParameters.resourceRichness
+        // passable land tiles (no mountains, no wonders) without resources yet
+        val candidateTiles = tileMap.values.filter { it.resource == null && it.isLand && !it.getLastTerrain().impassable }
+        val totalNumberOfResources = candidateTiles.size * tileMap.mapParameters.resourceRichness
         val resourcesPerType = (totalNumberOfResources/strategicResources.size).toInt()
         for (resource in strategicResources) {
-            val suitableTiles = tileMap.values
-                    .filter { it.resource == null && resource.terrainsCanBeFoundOn.contains(it.getLastTerrain().name) }
+            // remove the tiles where previous resources have been placed
+            val suitableTiles = candidateTiles
+                    .filter { it.resource == null && resource.terrainsCanBeFoundOn.contains(it.getBaseTerrain().name)}
 
             val locations = chooseSpreadOutLocations(resourcesPerType, suitableTiles, distance)
 
-            for (location in locations) location.resource = resource.name
+            for (location in locations) {
+                // clean features to be able to place the resource
```

Comment:
I think that we can keep the terrain feature, actually
Having forest on top of gold is fine - it means that the forest will need to be cleared in order to access the gold, but that's OK, and I think it looks more 'real' to have the vegetation 'ignore' the fact that there are resources there and grow anyway =)

Reply:
For such resources like gold the check `resource.terrainsCanBeFoundOn.contains(location.getLastTerrain().name)` will be `true` and it will not clear the forest, so that's ok in this case.

Reply:
I`ve just checked the `TileResources.json` and surprisingly find that Forest is not in `terrainsCanBeFoundOn` list for many fossils. I have added it to them (except Coal and Oil since they are products of wood's decay, and can't be under the forest).

Reply:
Maybe I'm missing something.
![image](https://user-images.githubusercontent.com/8366208/77847717-e1dcdc80-71c7-11ea-93c8-eed99f3d138f.png)
Forest is the last terrain, in this case, and it's not in terrains.canBeFoundOn
I don't think that adding forest and jungle to all the various resources is a good way of solving this either

Reply:
It is just added in the commit below.
https://github.com/yairm210/Unciv/pull/2270/commits/4e1fcd8c45ffd08090ce4b808d228636bfccfc6b#diff-3b9ba042d494afbd5af771f7a8c009feR172

> I don't think that adding forest and jungle to all the various resources is a good way of solving this either

Hmm... that seems to me natural to say explicitly where the resource can be found or not. Any better ideas?

Reply:
When the resource can be found on the base terrain, even if there's overgrowth (jungle, marsh, forest), we should allow the resource to be found there

Reply:
Well, then the fix becomes more simple.
I have reverted not necessary changes.